### PR TITLE
Update guava

### PR DESCRIPTION
**Pull request by o-liver**
Current newest version of guava is 29.0-jre

### CODE REVIEW

**Comment by o-liver** [comment]
Many tests are skipped and I don't know why...

**Comment by 1084565** [comment]
@marcusholl ? Could you please have a look?

**Comment by marcusholl** [comment]
@o-liver 

>Many tests are skipped and I don't know why...

I checked for two test files (`CMODataClientFileUploadTest`, `CMODataClientCreateTransportTest`). The number of test runs minus the number of skipped test equals the number of tests in the files. Don't know why the number of test runs is bigger than the number of tests in the file, which is "fixed" by having skipped tests. We should try to understand that.

Anyway this is the same for the current `master`. From that point of view can merge this PR here.

```
mvn clean package 2>&1 |grep -i skipped |grep -v Time
Tests run: 43, Failures: 0, Errors: 0, Skipped: 12
Tests run: 11, Failures: 0, Errors: 0, Skipped: 1
Tests run: 99, Failures: 0, Errors: 0, Skipped: 2
```

File: pom.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>18.0</version>
+            <version>29.0-jre</version>
```
Comment (marcusholl) [inline pom.xml:99]: The `-jre` version requires java 8. The `-android` version allows java 7. From the compatibility point of view `-android` would be better. On the other hand we can't stick to old versions all the time.

When merging this.
- For the next release we should have a hint that we require java 8.
- Maybe we should make the required java version more transparent by introducing a mvn enforcer rule for that.

**Pull request merged**